### PR TITLE
Upgrade 4 Integration

### PR DESCRIPTION
**Pull request by RachelFranks**
Adds support for ArbOS `upgrade-4` by
- Interpreting new gas accounting tuple
- Ensuring sequence number return codes are possible for the given ArbOS version
- Removing out of funds test case that incorrectly measures the proportion of fees sent to aggregators
- Future-proofing chain parameters set by having them come from the 0 address
- And other trivial test tweaks

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -3424,6 +3424,11 @@
   dependencies:
     ethers "^5.0.2"
 
+"@typechain/ethers-v5@^6.0.1":
```
Comment (hkalodner) [inline yarn.lock:3427]: I think you didn't have the latest arbos checked out when you last ran `yarn`. I don't believe there should be any `yarn.lock` diff in this PR

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: packages/arb-rpc-node/arbostest/fee_test.go [hunk 2]
```diff
@@ -100,20 +100,14 @@ type txTemplate struct {
...
-		t.Log("Using old calldata accounting")
-		countCalldataFunc = countCalldataUnitsOld
-		l1GasPerL2Calldata = big.NewInt(16)
-	} else {
-		t.Log("Using new calldata accounting")
-		countCalldataFunc = countCalldataUnitsNew
-		l1GasPerL2Calldata = big.NewInt(1)
-	}
+	t.Log("Using new calldata accounting")
+	countCalldataFunc = countCalldataUnitsNew
```
[the remark this answers is not in the document]
Comment (RachelFranks) [reply packages/arb-rpc-node/arbostest/fee_test.go:109]: Good call, we now use `countCalldataUnits` as an inline function.

File: packages/arb-rpc-node/arbostest/fee_test.go [hunk 3]
```diff
@@ -204,18 +198,6 @@ func TestFees(t *testing.T) {
 			nonzeroComputation: []bool{true},
 			correctStorageUsed: 0,
 		},
-		// Reverted storage allocating function call
```
[the remark this answers is not in the document]
Comment (RachelFranks) [reply packages/arb-rpc-node/arbostest/fee_test.go:207]: I'm actually not sure how that got removed. I've added it back in and the tests still pass.

File: yarn.lock (hunk 1 above)
Comment (RachelFranks) [reply yarn.lock:3427]: Added

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**